apps/meteor/app/lib/server/methods/sendMessage.ts

```diff
@@ -81,6 +81,14 @@ export async function executeSendMessage(uid: IUser['_id'], message: AtLeast<IMe
 	try {
 		const room = await canSendMessageAsync(rid, { uid, username: user.username, type: user.type });
 
+		if (room.encrypted && !settings.get<boolean>('E2E_Allow_Unencrypted_Messages')) {
```

Comment:
Can you please add a test case on API end-to-end tests to validate this error? One case for when it shouldn't be thrown and another one for when it should.
I believe we can do this in [here](https://github.com/RocketChat/Rocket.Chat/blob/3ed5bda6305a6b6a561720bbd60579766a817225/apps/meteor/tests/end-to-end/api/03-groups.js#L124)